src/peerstate.rs

```diff
@@ -320,9 +322,13 @@ impl<'a> Peerstate<'a> {
     pub fn render_gossip_header(&self, min_verified: PeerstateVerifiedStatus) -> Option<String> {
         if let Some(key) = self.peek_key(min_verified) {
             // TODO: avoid cloning
+            let public_key = match SignedPublicKey::try_from(key.clone()) {
+                Ok(key) => key,
```

Comment:
Use `ok`(https://doc.rust-lang.org/std/result/enum.Result.html#method.ok) and `?`. `?` works for `Option`.

Reply:
Oh nice.  Whenever I write a match arm that just unwraps something I feel like I'm doing something suboptimal.  Thanks for the hint.